### PR TITLE
feat: expose serve-static builder

### PR DESCRIPTION
**Pull request by cometkim**
This allows accessing `middleware/serve-static` from third-party

### Use case

I run my own handler based on Cloudflare R2 to avoid the limitations of Cloudflare KV / Pages. And I found that most of the code I needed there was already built into hono.

So what I expect is like:

```ts
import { Hono } from 'hono';
import { serveStatic as makeServe } from 'hono/base-serve-static';

const app = new Hono<Env>();

const serveContent = makeServe<Env>({
  getContent: async (path, c) => {
    // retrieve
    let response = await cache.match(c.req.raw);

    // ...get R2 object
    if (!response) {
      let obj = await c.env.CONTENT.get(path);
      if (obj) {
        response = obj.body;
        c.executionCtx.waitUntil(cache.put(c.req.raw, obj.body));
      }
    }

    c.body(response);
  },
});

app.use(serveContent);

export default app;
```

### Author should do the followings, if applicable

- [ ] Add tests
- [x] Run tests
- [x] `yarn denoify` to generate files for Deno

### CODE REVIEW

**Comment by yusukebe** [comment]
Hi @cometkim 

This is good! But please wait until I look over the details!

**Comment by cometkim** [comment]
@yusukebe added a type definition rather than `any` as it would be a public interface.

And made some changes to allow a situation where users needed to create the Response object themselves. (e.g. using with Browser/Cloudflare's Cache API)

File: package.json [hunk 1]
```diff
@@ -279,6 +279,11 @@
       "import": "./dist/helper/factory/index.js",
       "require": "./dist/cjs/helper/factory/index.js"
     },
+    "./base-serve-static": {
```
Comment (yusukebe) [inline package.json:282]: Simply, I think `./base-serve` is better than `./base-serve-static`.

```ts
import { serveStatic } from 'hono/serve-static 
```

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
@cometkim Thanks!

**Pull request merged**